sympy/core/numbers.py

```diff
@@ -1879,9 +1880,11 @@ def __ceil__(self):
         return self.ceiling()
 
     def __eq__(self, other):
-        try:
-            other = _sympify(other)
-        except SympifyError:
+        if isinstance(other, MPQ):
+            return self.p == other.numerator and self.q == other.denominator
+        if isinstance(other, (int, float)):
```

Comment:
Won't this lead to that?

Rational(10000000000000001, 10000000000000000) == 1.0

Reply:
Yes, I believe it would, which would be incorrect. I'll also add a test for this, so it won't be possible to accidentally introduce this behaviour in the future.

Reply:
Probably better to compare `self.p == other*self.q`. For ints that should be clearly better, but for floats it may provide some other weird consequences.

Reply:
I tried that first (since it should be more efficient), but it caused issues with `== float('inf')`

Reply:
OK! But for int it should be better?